### DISCUSSION TITLE
How to define many to one dependencies in dagster

### DISCUSSION THREAD

**Discussion by dpeng817** [dagster-io/dagster#13870] [Q&A]
Original slack thread: https://dagster.slack.com/archives/C01U954MEER/p1682340313548489

Consider the situation where one asset lives downstream of many independent assets. How does one ergonomically define the dependency from those many assets to the one downstream asset?

**Comment by dpeng817** [comment]
If you don't need to access the outputs of the upstream assets, you can use [`non_argument_deps`](https://docs.dagster.io/concepts/assets/software-defined-assets#non-argument-dependencies) to define the many dependencies at once, without needing to clog up the inputs to your asset. 
If you need to use IO managers, the solution from [this thread](https://github.com/dagster-io/dagster/discussions/10008) could work

**Reply by clayheaton** [reply]
I do this in some jobs, however, I have other jobs with hundreds of independent assets and it is very cumbersome to list all of the assets.

**Comment by clayheaton** [comment]
One idea would be to enable tagging of assets and then have `non_argument_deps` accept a tag belonging to many assets as a dependency.

**Reply by dpeng817** [reply]
Another option could be specifying a dependency to an entire group of assets
